Sources/swift-algorand-sdk/v2/client/indexer/LookUpAssetTransactions.swift

```diff
@@ -7,168 +7,106 @@
-            self.queryItems["min-round"] = "\(uMinRound)"
-        }
-        if let uNext = next{
-            self.queryItems["next"] = uNext
+    private var queryItems: [String: String]=[:] {
+        didSet {
+            self.parameters.queryParameters = queryItems
         }
-        if let uNotePrefix = notePrefix{
-            self.queryItems["note-prefix"] = CustomEncoder.encodeToBase64(uNotePrefix);
-        }
-        if let uRekeyTo = rekeyTo{
-            self.queryItems["rekey-to"] = "\(uRekeyTo)"
-        }
-        if let uRound = round{
-            self.queryItems["round"] = "\(uRound)";
-        }
-        if let uTxId = txid{
-            self.queryItems["txid"] = uTxId
-        }
-        
     }
-    public func execute( callback: @escaping (_:Response<TransactionsResponse>)->Void) {
     
-        let headers:HTTPHeaders=[client.apiKey:client.token]
-//        print(getRequestString())
-        var request=AF.request(getRequestString(), method: .get, headers: headers,requestModifier: { $0.timeoutInterval = 120 })
-//        request.responseJSON(){response in
-//            debugPrint(response.value)
-//        }
-        var customResponse:Response<TransactionsResponse>=Response()
-      request.responseDecodable(of: TransactionsResponse.self){ (response) in
-        if(response.error != nil){
-            customResponse.setIsSuccessful(value:false)
-            var errorDescription=String(data:response.data ?? Data(response.error!.errorDescription!.utf8),encoding: .utf8)
-            customResponse.setErrorDescription(errorDescription:errorDescription!)
-            callback(customResponse)
-            if(response.data != nil){
-                if let message = String(data: response.data!,encoding: .utf8){
-                    var errorDic = try! JSONSerialization.jsonObject(with: message.data, options: []) as? [String: Any]
-                    customResponse.errorMessage = errorDic!["message"] as! String
-                
-                }
-
-            }
-            return
-        }
-                        let data=response.value
-                        var transactionsResponse:TransactionsResponse=data!
-                        customResponse.setData(data:transactionsResponse)
-                        customResponse.setIsSuccessful(value:true)
-                        callback(customResponse)
-
-        }
-        
+    init(client: IndexerClient,
+         assetId: Int64,
+         address: Address? = nil,
+         currencyGreaterThan: Int64? = nil,
+         currencyLessThan: Int64? = nil,
+         excludeCloseTo: Bool? = nil,
+         limit: Int64? = nil,
+         maxRound: Int64? = nil,
+         minRound: Int64? = nil,
+         next: String? = nil,
+         notePrefix: Data? = nil,
+         rekeyTo: Bool? = nil,
+         round: Int64? = nil,
+         txid: String? = nil) {
+        self.client = client
+        parameters = .init(path: "/v2/assets/\(assetId)/balances")
```

Comment:
Please Is queryItems not supposed to be initialised somewhere here?

Reply:
that's definetly a mistake. A strange one, because i remember implementing it... fixing it, thank you for spotting this :)